test/yz_index_hashtree_eqc.erl

```diff
@@ -0,0 +1,508 @@
+                                  S#state.yz_idx_tree, S#state.kv_idx_tree]}.
+
+%% @doc insert_both_pre - Precondition for insert_both
+-spec insert_both_pre(S :: eqc_statem:symbolic_state(),
+                      Args :: [term()]) -> boolean().
+insert_both_pre(S, _Args) ->
+    S#state.yz_idx_tree /= undefined andalso S#state.kv_idx_tree /= undefined.
+
+%% @doc insert_both_next - Next state function
+-spec insert_both_next(S :: eqc_statem:symbolic_state(),
+                       V :: eqc_statem:var(),
+                       Args :: [term()]) -> eqc_statem:symbolic_state().
+insert_both_next(S, _V, [_, RObj, _, _]) ->
+    {ok, YZTreeData} = dict:find({0, 1}, S#state.yz_idx_objects),
+    {ok, KVTreeData} = dict:find({0, 1}, S#state.kv_idx_objects),
+    S#state{yz_idx_objects=dict:store({0, 1},
+                                      set_treedata(RObj, YZTreeData),
+                                      S#state.yz_idx_objects),
+            kv_idx_objects=dict:store({0, 1},
+                                      set_treedata(RObj, KVTreeData),
+                                      S#state.kv_idx_objects),
+            trees_updated=false}.
+
+%% @doc insert_both_post - Postcondition for insert_both
+-spec insert_both_post(S :: eqc_statem:dynamic_state(),
+                       Args :: [term()], R :: term()) -> true | term().
+insert_both_post(_S, _Args, _Res) ->
+    true.
+
+-spec insert_both(sync|async, obj(), {ok, tree()}, {ok, tree()}) -> {ok, ok}.
+insert_both(Method, RObj, YZOkTree, KVOkTree) ->
+    {insert_yz_tree(Method, RObj, YZOkTree),
+     insert_kv_tree(Method, RObj, KVOkTree)}.
+
+%% ------ Grouped operator: update
+%% @doc update_command - Command generator
+-spec update_command(S :: eqc_statem:symbolic_state()) ->
+        eqc_gen:gen(eqc_statem:call()).
+update_command(S) ->
+    {call, ?MODULE, update, [S#state.yz_idx_tree, S#state.kv_idx_tree]}.
+
+%% @doc update_pre - Precondition for update
+-spec update_pre(S :: eqc_statem:symbolic_state(),
+                 Args :: [term()]) -> boolean().
+update_pre(S, _Args) ->
+    S#state.yz_idx_tree /= undefined andalso S#state.kv_idx_tree /= undefined.
+
+%% @doc update_next - Next state function
+-spec update_next(S :: eqc_statem:symbolic_state(),
+                  V :: eqc_statem:var(),
+                  Args :: [term()]) -> eqc_statem:symbolic_state().
+update_next(S, _Value, _Args) ->
+    S#state{trees_updated=true}.
+
+%% @doc update_post - Postcondition for update
+-spec update_post(S :: eqc_statem:dynamic_state(),
+                  Args :: [term()], R :: term()) -> true | term().
+update_post(_S, _Args, _Res) ->
+    true.
+
+%% @doc Update (commit) hashtrees.
+-spec update({ok, tree()}, {ok, tree()}) -> ok.
+update({ok, YZTreePid}, {ok, KVTreePid}) ->
+    yz_index_hashtree:update({0, 1}, YZTreePid),
+    riak_kv_index_hashtree:update({0, 1}, KVTreePid),
+    ok.
+
+%% ------ Grouped operator: compare
```

Comment:
Unless I'm missing something, or unless it's performance-prohibitive, shouldn't `compare` be integrated as an invariant rather than a distinct command?